pkg/vere/main.c

```diff
@@ -65,17 +65,31 @@ _main_self_path(void)
 
 /* _main_readw(): parse a word from a string.
 */
-static u3_noun
+static c3_t
 _main_readw(const c3_c* str_c, c3_w max_w, c3_w* out_w)
 {
   c3_c* end_c;
   c3_w  par_w = strtoul(str_c, &end_c, 0);
 
   if ( *str_c != '\0' && *end_c == '\0' && par_w < max_w ) {
     *out_w = par_w;
-    return c3y;
+    return 0;
   }
-  else return c3n;
+  else return 1;
+}
```

Comment:
I'll first mention that I included this commit in the PR because the use of `c3_o` for non-Urbit control flow bothered me so I wanted to force a discussion on the matter. However, the commit has nothing to do with `--cue` and I'm prepared to pull it out into a separate PR if the discussion runs too long.

> you've created a situation in which the if branch of the following code is taken when _main_readw() fails and the else branch is taken when _main_read() succeeds, which is directly opposite the convention in C

This is incorrect. It's even in the C standard for the main function that a non-zero return value indicates that an error has occurred [7.20.4.3.5](https://www.open-std.org/jtc1/sc22/WG14/www/docs/n1256.pdf). It's also a POSIX standard that `0` means success (no link; it costs money and is tedious to access).

If you look over the changes in the commit, you will see that all of the cases in/of `_main_readw()` and `_main_getopt()` were already using `1` as an error state: `c3n` (i.e. `1`) meant failure; `c3y` (i.e. `0`) meant success. Depending on how you look at it, they were actually *breaking* the C/POSIX convention, since the "false" loobean was being used to signal error.

Regardless of the above, using `c3_o` as part of the control flow for Vere is a bad idea that should be done away with. Loobeans should only be used in the cases where Vere is directly generating / operating on Urbit nouns. In the cases where loobean values from Urbit determine Vere control flow, we already have macros for converting loobeans to booleans.

Reply:
I appreciate you forcing the issue because I agree that `c3_o` should not be used except when interacting directly with nouns; I'm in favor of replacing all uses of `c3_o` that don't interact with nouns with `c3_t` or similar. If we're going to do so, we should do so methodically, changing from `c3_o` to `c3_t` in a file at a time, with an accompanying PR for each file changed. This conversation quickly turns into a conversation about Vere handles errors, which means we should probably adopt `errno` (or `errno`-like conventions) for returning errors.

With respect to `0` indicating success: yes, but not in the case when that `0` is a boolean (i.e. `bool` is used via `stdbool.h`). From my perspective, a non-zero, non-negative return value only indicates failure when returned from `main()`. In all other cases, I expect a non-zero, non-negative return value to indicate success of some sort with a negative return value reserved for failures. All this to say, I hear where you're coming from, and I don't disagree, but I do think the context here doesn't clearly indicate that `1` is actually failure (unless of course you were to document the functions indicating as much). `-1`, on the other hand, would make it more clear, at least to me, that `return -1` is a failure case.

Reply:
Introducing `errno` semantics would be great as part of improving the Vere error logging, but too big for what I'm attempting here. However, I think that this proposed commit is still valuable as an incremental improvement.

How would you feel about:
1. Moving this commit to its own PR
2. Replacing `1` with `-1`
3. Possibly replacing `c3_t` with another type like `c3_ws` or `c3_i` (since overloading `c3_t` with multiple `true` values would be out of line with the intent of the type)

?

Reply:
Agreed about `errno` semantics on both points. (1)-(3) sound like a good plan. I'd prefer `c3_i` over `c3_ws` since it's in line with POSIX/Linux.